languagetool-language-modules/de/src/main/java/org/languagetool/rules/de/GermanSpellerRule.java

```diff
@@ -2339,6 +2354,231 @@ private boolean isOnlyPluralNoun(String word) throws IOException {
     return isPluralNoun(word) && word.endsWith("en") && isOnlyNoun(word);
   }
 
+  private boolean processTwoPartCompounds(String part1, String part2) throws IOException {
+    // Expects two parts of a compound and checks
+    //  if their POS tags are correct,
+    //  if an infix s is missing or
+    //  if an infix s is incorrect
+    String part1uc = uppercaseFirstChar(part1);
+    String part2uc = uppercaseFirstChar(part2);
+    String part1WithoutHyphen = removeHyphen(part1);
+    boolean part2ucIsNoun = isNoun(part2uc);
+    boolean part2ucIsMisspelled = isMisspelled(uppercaseFirstChar(part2uc));
```

Comment:
🥜 `uc` is not really transparent, maybe just use `upcased` for better readability.

Reply:
You're right. I adopted this naming convention when I started working on this task^^